stix2/test/test_bundle.py

```diff
@@ -158,3 +158,11 @@ def test_parse_unknown_type():
     with pytest.raises(stix2.exceptions.ParseError) as excinfo:
         stix2.parse(unknown)
     assert str(excinfo.value) == "Can't parse unknown object type 'other'! For custom types, use the CustomObject decorator."
+
+
+def test_stix_object_property():
+    prop = stix2.core.STIXObjectProperty()
+
+    identity = stix2.Identity(name="test", identity_class="individual")
+    assert prop.clean(identity) == identity
+    assert prop.clean(identity) is identity
```

Comment:
We don't need both `==` and `is` checks.

Reply:
In theory, this could be appropriate, if you wanted to test an `__eq__` override and make sure it's consistent with `is`.  But the stix objs don't have that override do they?  So, perhaps unnecessary in this case.  An `__eq__` could be defined somewhere perhaps, which checks the `id` and `modified` properties of stix objects.  Then a test like this could make sense I think.

Reply:
I'm pretty sure `is` ensures the values point to the same item in memory. I believe we get `__eq__` for free with `collections.abc.Mapping`, but it's not as useful for tests since you would need to ensure even the auto-generated properties like `id` and `created` are the same.

Each tests has its purpose, but in this case I'm not sure which is more important, that `prop.clean` returns the same object, or just an equivalent object.

Reply:
This test is making sure that instances of our STIX classes get added to Bundles without being re-parsed/dictified: https://github.com/oasis-open/cti-python-stix2/pull/88/commits/374539a6cc7fe0f380ab598ed39d791967119afb#diff-0b92af30cf45a78c5cf9a35dd394cfeaR22

Thus I think the `is` check is the correct one in this case.